docs/4.x/tracking-java.md

```diff
@@ -0,0 +1,169 @@
+
+## Authentikation Token
+
+The parameter `token_auth` contains the authentication token of the Matomo user. Some parameters require it. The authorization key can be found in the Matomo user interface under "Administration" -> "Personal" -> "Security" -> "Auth tokens"
+
+Using the Matomo Java Tracker are three different ways to set the authentication token:
+
+1. Set the authentication token globally in the tracker configuration. The authentication token will be used for all tracking requests. The authorization key can be set using the method `TrackerConfiguration.builder().defaultTokenAuth(String)`.
+
+2. Set the authentication token in the tracking request. The authentication token will be used for this tracking request only. The authorization key can be set using the method `MatomoRequest.request().authToken(String)`.
+
+3. (Deprecated) Call a send method of the tracker with the authentication token as a parameter. The authentication token will be used for this tracking request only. Since this method is redundant, it is deprecated and will be removed in a future version. It has the same effect than setting it in the request. Set it in the request or globally instead.
+
+The tracker uses primarily the authentication set via method parameter, then the authentication set in the request, and finally the authentication set in the tracker configuration. If no authentication token is set, the tracker won't set the parameter `token_auth` in the tracking request. This can lead to problems if you use parameters that require an authentication token. In this case, the tracker will throw an exception on sending the request.
+
+Please mind that the token should be transmitted securely, for example using HTTPS. 
+
+## Custom Tracking Parameters
+
+You can also set additional custom request parameters using the method `MatomoRequest.request().additionalParameters(Map<String, Collection<Object>>)`. The additional parameters will be added to the request URL as query parameters. The query parameter name will be the key of the map entry and the query parameter value will be the value of the map entry. To add the same query parameter name with multiple values, you can use a collection as the parameter value, e.g. `additionalParameters(Map.of("foo", List.of("bar", "baz")))` will add the query parameters `?foo=bar&foo=baz` to the request URL (note that the order of the query parameters is not guaranteed). Note: If you set any parameter that is not supported by the Matomo tracking API, the parameter will be ignored by Matomo, although it will be added to the request URL.
```

Comment:
> `additionalParameters(Map.of("foo", List.of("bar", "baz")))` will add the query parameters `?foo=bar&foo=baz`

What's the purpose of setting the same parameter multiple times? Wouldn't always the last value be used in that case?
Or should that be something like `?foo[]=bar&foo[]=baz` to allow sending array values?

Reply:
In that case it's "fine". But maybe you should consider changing that in the java tracker with the next major release or so...

Reply:
I opened a ticket: https://github.com/matomo-org/matomo-java-tracker/issues/163